openpype/hosts/fusion/plugins/create/create_saver.py

```diff
@@ -119,23 +117,26 @@ def _imprint(self, tool, data):
 
     def _update_tool_with_data(self, tool, data):
         """Update tool node name and output path based on subset data"""
-        if "subset" not in data:
+
+        if data.get("subset") is None:
+            self.log.warning("No subset found for _update_tool_with_data")
             return
 
-        original_subset = tool.GetData("openpype.subset")
+        # Get current session but change the asset to the one from
+        # data in case a new asset has bin selected
+        session = legacy_io.Session
```

Comment:
Please don't change `legacy_io.Session` for other purposes than real change of context. Also we're trying to remove `legacy_io` and `Session`. Rather use other ways how to get workdir path please.

Reply:
Also this is producing a lot of calls to databases on save. I can imagine with 10 and more instances it could take tens of seconds.